### PR TITLE
refactor(lsp): deprecate util methods

### PR DESCRIPTION
**Pull request by MariaSolOs**
Let the culling of `vim.lsp.util` begin.

Part of #25272.

### CODE REVIEW

File: runtime/doc/deprecated.txt [hunk 1]
```diff
@@ -142,6 +142,8 @@ LSP FUNCTIONS
 - *vim.lsp.get_active_clients()*		Use |vim.lsp.get_clients()|
 - *vim.lsp.for_each_buffer_client()*		Use |vim.lsp.get_clients()|
 - *vim.lsp.util.trim_empty_lines()*		Use |vim.split()| with `trimempty` instead.
+- *vim.lsp.util.try_trim_markdown_code_blocks()*
```
Comment (MariaSolOs) [inline runtime/doc/deprecated.txt:145]: Thoughts on not mentioning alternatives here?

Comment (justinmk) [reply runtime/doc/deprecated.txt:145]: No objection.

File: runtime/lua/vim/lsp/util.lua [hunk 2]
```diff
@@ -1991,6 +1991,7 @@ end
 ---
 --- CAUTION: Modifies the input in-place!
 ---
+---@deprecated
 ---@param lines table list of lines
 ---@return string filetype or "markdown" if it was unchanged.
```
Comment (justinmk) [inline runtime/lua/vim/lsp/util.lua:1996]: If the deprecated functions are not used in this codebase then we can go ahead and delete their docs too.
```suggestion
---@deprecated
```

**Review by justinmk — APPROVED** [review]

File: runtime/lua/vim/lsp/util.lua (hunk 2 above)
Comment (MariaSolOs) [reply runtime/lua/vim/lsp/util.lua:1996]: I prefer to leave the docs for now and delete them together with the code in the future. The type annotations also help LuaLS (type warnings give me anxiety).

Comment (clason) [reply runtime/lua/vim/lsp/util.lua:1996]: For the sake of completeness: We _could_ keep just the annotations and remove the descriptions, but I agree that while the code is there, the full documentation is useful for code archeology (e.g., no need to hunt further for the commit _with_ description).

**Review by gpanders — APPROVED** [review]
🔥 🔥 🔥

**Pull request merged**